src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -626,26 +627,48 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
     return Utils.arrayEqual(this.activePath, this.options.basePath);
   }
 
+  private buildClearIcon() {
```

Comment:
I'm seeing this file is quite long. I would prefer to extract code into a CategoryFacetHeader.ts file, that would contain the clear button and the title section.

Even better might be to extract the eraser code out of FacetHeader.ts file and convert it to a component that can accept onClick method, logAnalytics method etc. If that is too complicated, then just extracting could be enough for now.

Reply:
+1